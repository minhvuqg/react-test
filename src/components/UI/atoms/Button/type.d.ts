import React from 'react'

export interface IProps
  extends React.DetailedHTMLProps<
      React.ButtonHTMLAttributes<HTMLButtonElement>,
      HTMLButtonElement
    >,
    React.AriaAttributes {
  children: string | React.ReactNode
  primary: boolean
}
