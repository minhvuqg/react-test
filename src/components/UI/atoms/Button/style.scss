@import 'scss/var/color.scss';

.-custom-btn {
  height: 40px;
  max-width: 250px;
  border: none;
  padding: 0 16px;
  border-radius: 8px;
  cursor: pointer;

  &.-primary {
    color: #fff;
    transition: all 0.3s;
    background-color: $primary-color;

    &:hover {
      background-color: $primary-color-hover;
    }
  }
}
