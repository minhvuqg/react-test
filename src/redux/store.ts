import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
