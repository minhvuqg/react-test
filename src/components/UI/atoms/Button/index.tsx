import { FC } from 'react'
import { IProps } from './type.d'

import './style.scss'

const primaryClasses = "-primary"

const Button: FC<IProps> = ({
  children,
  className,
  primary = false,
  ...props
}: IProps): JSX.Element => {
  return (
    <button className={`-custom-btn ${className} ${primary ? primaryClasses : ""}`} {...props}>
      {children}
    </button>
  )
}

export { Button }
