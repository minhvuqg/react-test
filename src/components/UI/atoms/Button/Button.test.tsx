import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { Button } from '../Button'
import store from 'redux/store'
import { BrowserRouter as Router } from 'react-router-dom'

it('render primary button correctly', () => {
  const primaryButton = TestRenderer.create(
    < Provider store={store} >
      <Router>
        <Button data-testid="addToCartBtn"
          children={"Label"}
          primary={true}
          onClick={() => { }} />
      </Router>
    </Provider >).toJSON()
  expect(primaryButton).toMatchSnapshot()
})

it('render button correctly', () => {
  const primaryButton = TestRenderer.create(
    < Provider store={store} >
      <Router>
        <Button data-testid="addToCartBtn"
          children={"Label"}
          primary={false}
          onClick={() => { }} />
      </Router>
    </Provider >).toJSON()
  expect(primaryButton).toMatchSnapshot()
})
