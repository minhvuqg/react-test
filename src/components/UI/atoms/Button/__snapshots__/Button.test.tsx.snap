// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button correctly 1`] = `
<button
  className="-custom-btn undefined "
  data-testid="addToCartBtn"
  onClick={[Function]}
>
  Label
</button>
`;

exports[`render primary button correctly 1`] = `
<button
  className="-custom-btn undefined -primary"
  data-testid="addToCartBtn"
  onClick={[Function]}
>
  Label
</button>
`;
